app/src/main/java/fr/free/nrw/commons/quiz/QuizActivity.java

```diff
@@ -0,0 +1,168 @@
+    @BindView(R.id.question_text) TextView questionText;
+    @BindView(R.id.question_title) TextView questionTitle;
+    @BindView(R.id.quiz_positive_answer) RadioButton positiveAnswer;
+    @BindView(R.id.quiz_negative_answer) RadioButton negativeAnswer;
+    @BindView(R.id.toolbar) Toolbar toolbar;
+
+    private QuizController quizController = new QuizController();
+    private ArrayList<QuizQuestion> quiz = new ArrayList<QuizQuestion>();
+    private int questionIndex = 0;
+    private int score;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_quiz);
+        Fresco.initialize(this);
+
+        quizController.initialize(this);
+        ButterKnife.bind(this);
+        setSupportActionBar(toolbar);
+        displayQuestion();
+
+    }
+
+    /**
+     * to move to next question and check whether answer is selected or not
+     */
+    @OnClick(R.id.next_button)
+    public void setNextQuestion(){
+        if( questionIndex <= quiz.size() && (positiveAnswer.isChecked() || negativeAnswer.isChecked())) {
+            evaluateScore();
+        } else if ( !positiveAnswer.isChecked() && !negativeAnswer.isChecked()){
+            AlertDialog.Builder alert = new AlertDialog.Builder(this);
+            alert.setTitle(getResources().getString(R.string.warning));
+            alert.setMessage(getResources().getString(R.string.warning_for_no_answer));
+            alert.setPositiveButton("Continue", new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    dialog.dismiss();
+                }
+            });
+            AlertDialog dialog = alert.create();
+            dialog.show();
+        }
+
+    }
+
+    /**
+     * to give warning before ending quiz
+     */
+    @Override
+    public void onBackPressed() {
+        AlertDialog.Builder alert = new AlertDialog.Builder(this);
+        alert.setTitle(getResources().getString(R.string.warning));
+        alert.setMessage(getResources().getString(R.string.quiz_back_button));
+        alert.setPositiveButton("Continue", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                Intent i = new Intent(QuizActivity.this, QuizResultActivity.class);
+                dialog.dismiss();
+                i.putExtra("QuizResult",score);
+                startActivity(i);
+            }
+        });
+        alert.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialogInterface, int i) {
+                dialogInterface.dismiss();
+            }
+        });
+        AlertDialog dialog = alert.create();
+        dialog.show();
+        //super.onBackPressed();
```

Comment:
- Do you want to keep the commented-out line?
- Is there no way to return back from this activity. I haven't tested it on a real device yet so am not sure how the behavior might be.